app/controllers/pxe_controller.rb

```diff
@@ -61,52 +61,15 @@ def explorer
     @breadcrumbs = []
     @explorer = true
     @sb[:open_tree_nodes] ||= []
-    @trees = []
-    @accords = []
-
-    if role_allows(:feature => "pxe_server_accord", :any => true)
-      self.x_active_tree ||= 'pxe_servers_tree'
-      self.x_active_accord ||= 'pxe_servers'
-      @trees << pxe_server_build_tree
-      @accords.push(:name      => "pxe_servers",
-                    :title     => "PXE Servers",
-                    :container => "pxe_servers_accord",
-                    :image     => "pxeserver")
-    end
-
-    if role_allows(:feature => "customization_template_accord", :any => true)
-      self.x_active_tree ||= 'customization_templates_tree'
-      self.x_active_accord ||= 'customization_templates'
-      @trees << customization_template_build_tree
-      @accords.push(:name      => "customization_templates",
-                    :title     => "Customization Templates",
-                    :container => "customization_templates_accord",
-                    :image     => "customizationtemplate")
-    end
-
-    if role_allows(:feature => "pxe_image_type_accord", :any => true)
-      self.x_active_tree ||= 'pxe_image_types_tree'
-      self.x_active_accord ||= 'pxe_image_types'
-      @trees << pxe_image_type_build_tree
-      @accords.push(:name      => "pxe_image_types",
-                    :title     => "System Image Types",
-                    :container => "pxe_image_types_accord",
-                    :image     => "pxeimagetype")
-    end
 
-    if role_allows(:feature => "iso_datastore_accord", :any => true)
-      self.x_active_tree ||= 'iso_datastores_tree'
-      self.x_active_accord ||= 'iso_datastores'
-      @trees << iso_datastore_build_tree
-      @accords.push(:name      => "iso_datastores",
-                    :title     => "ISO Datastores",
-                    :container => "iso_datastores_accord",
-                    :image     => "isodatastore")
-    end
+    # Build the Explorer screen from scratch
+    allowed_features = ApplicationController::Feature.allowed_features(features)
+    @trees = allowed_features.collect { |feature| feature.build_tree(@sb) }
+    @accords = allowed_features.map(&:accord_hash)
+    set_active_elements(allowed_features.first)
```

Comment:
i can move these into single method as a follow up PR or in same PR as another commit, let me know.